### PR TITLE
Implement isin()

### PR DESCRIPTION
**Pull request by Dhruv454000**
## Description

Implement isin() functionality in heat.

Issue/s resolved: #450

## Type of change

- New feature 

## Due Diligence
- [ ] All split configurations tested
- [ ] Multiple dtypes tested in relevant functions
- [ ] Documentation updated (if needed)
- [ ] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
 no

### CODE REVIEW

**Comment by mtar** [comment]
GPU cluster tests are currently disabled on this Pull Request.

**Comment by mrfh92** [comment]
see #450 - this PR, however, is as far as I see only concerned with `in1d` only. 
Can serve as a starting point to implement `isin()` also. 

(Reviewed within #1109 )

File: heat/core/logical.py [hunk 1]
```diff
@@ -32,6 +33,7 @@
     "logical_or",
     "logical_xor",
     "signbit",
+    "in1d",
```
Comment (ClaudiaComito) [inline heat/core/logical.py:36]: this list should be in alphabetical order

File: heat/core/logical.py [hunk 2]
```diff
@@ -547,3 +549,88 @@ def signbit(x: DNDarray, out: Optional[DNDarray] = None) -> DNDarray:
...
+    Test whether each element of a 1-D array is also present in a second array.
+    Returns a boolean array the same length as `x` that is True
+    where an element of `x` is in `y` and False otherwise.
+    We recommend using :func:`isin` instead of `in1d` for new code.
+
+    Parameters
+    ----------
+    x : DNDarray or scalar
+        Input array.
+    y : DNDarray or scalar
```
Comment (ClaudiaComito) [inline heat/core/logical.py:570]: We should implement `isin`, and then alias `in1d` to a special case of `isin`.

We can use the generic  `_operations.binary_op` function with `torch.isin` as local function, there's no need to implement the algorithm from scratch.

**Review by ClaudiaComito — CHANGES_REQUESTED** [review]
@Dhruv454000 I hope you're well. I have to apologize, this PR got buried somehow. If you're still interested and have time, I can help you finish it. 

If we don't hear from you within two weeks, I'll finish the PR. Thanks again for your time!

**Pull request closed without merging**